src/benchmarks/gc/src/commonlib/parse_and_serialize.py

```diff
@@ -294,24 +294,25 @@ def write_yaml_file(path: Path, content: object, overwrite: bool = False) -> Non
         dump(serializable, f, Dumper=MyDumper, default_flow_style=False)
 
 
-def _format_result_yaml_fields(content: OrderedDict, indent_size: int, result: [str]) -> None:
-    for k, v in content:
-        if isinstance(v, Dict):
-            text = f"{k}:\n"
-            result.append('{field: >{width}}'.format(field=text, width=len(text) + indent_size))
-            _format_result_yaml_fields(v.items(), indent_size + 2, result)
-        else:
-            if (k == 'stdout'):
-                v = v.rstrip("\n").replace("\n", "\n  ")
-                text = f"{k}:\n  \"{v}\"\n"
+def _format_result_yaml_fields(content: object, indent_size: int, result: List[str]) -> None:
+    if isinstance(content, OrderedDict):
+        for k, v in content.items():
+            if isinstance(v, Dict):
+                text = f"{k}:\n"
+                result.append("{field: >{width}}".format(field=text, width=len(text) + indent_size))
+                _format_result_yaml_fields(v.items(), indent_size + 2, result)
             else:
-                text = f"{k}: {v}\n"
-            result.append('{field: >{width}}'.format(field=text, width=len(text) + indent_size))
+                if k == "stdout":
+                    v = v.rstrip("\n").replace("\n", "\n  ")
+                    text = f'{k}:\n  "{v}"\n'
+                else:
+                    text = f"{k}: {v}\n"
+                result.append("{field: >{width}}".format(field=text, width=len(text) + indent_size))
 
 
 def write_test_yaml_file(path: Path, content: object) -> None:
-    yaml_items = []
-    _format_result_yaml_fields(to_serializable(content).items(), 0, yaml_items)
+    yaml_items: List[str] = []
```

Comment:
what was the error before? it'd be helpful to show the errors/warnings that the linter detects when we make changes like these.